packages/jest-dom-mocks/src/dimension.ts

```diff
@@ -0,0 +1,68 @@
+enum SupportedDimension {
+  OffsetWidth = 'offsetWidth',
+  OffsetHeight = 'offsetHeight',
+  ScrollWidth = 'scrollWidth',
+  ScrollHeight = 'scrollHeight',
+}
+
+type DimensionMap = {[T in SupportedDimension]: HTMLElement | Element};
+type MockedDimensions = {[T in SupportedDimension]: number};
+
+export default class Dimension {
+  private isUsingMock = false;
+  private mocks: Partial<MockedDimensions> = {};
+  private dimensionMap: DimensionMap = {
```

Comment:
It is probably safer not to actually reference `Element`/ `HTMLElement` until after `mock` is called, otherwise just initializing this can crash tests that bring in one of the other mocks but aren't using the browser jest environment.